protocol-designer/src/components/organisms/EditLabwareQuantityModal/index.tsx

```diff
@@ -0,0 +1,145 @@
+import { useEffect, useState } from 'react'
+import { createPortal } from 'react-dom'
+import { useTranslation } from 'react-i18next'
+import { useDispatch, useSelector } from 'react-redux'
+
+import {
+  COLORS,
+  DIRECTION_COLUMN,
+  Flex,
+  InputField,
+  JUSTIFY_END,
+  Modal,
+  PrimaryButton,
+  SecondaryButton,
+  SPACING,
+  StyledText,
+} from '@opentrons/components'
+
+import {
+  createContainer,
+  deleteContainer,
+} from '../../../labware-ingred/actions'
+import { getLabwareEntities } from '../../../step-forms/selectors'
+import { HandleEnter } from '../../atoms'
+import { getMainPagePortalEl } from '../Portal'
+
+import type { ThunkDispatch } from '../../../types'
+
+interface EditLabwareQuantityModalProps {
+  allLabwareIdsOnStack: string[]
+  labwareId: string
+  onClose: () => void
+}
+export function EditLabwareQuantityModal(
+  props: EditLabwareQuantityModalProps
+): JSX.Element {
+  const { onClose, allLabwareIdsOnStack, labwareId } = props
+  const { t } = useTranslation(['starting_deck_state', 'shared'])
+  const dispatch = useDispatch<ThunkDispatch<any>>()
+  const labwareEntities = useSelector(getLabwareEntities)
+  const { labwareDefURI, def } = labwareEntities[labwareId]
+  const stackingLimit = def.stackLimit ?? 0
+  const initialQuantity = allLabwareIdsOnStack.length
+  const [quantity, setQuantity] = useState<string>(initialQuantity.toString())
```

Comment:
maybe

```suggestion
  const [quantity, setQuantity] = useState<number>(initialQuantity)
```

using `parseInt` in InputField and removing other `parseInt` and quantity === ''.
i guess comparing number and easy and safe.

Reply:
i changed it to a string because it is used in the input field 😅